R/utils_render_register_json.R

```diff
@@ -108,12 +94,22 @@ render_register_json <- function(filter, register_table, register_table_name) {
 #' 
 #' @param list_register_table List of register tables
 render_register_jsons <- function(list_register_tables){
-  # Loop over each register table and render a json
   for (filter in names(list_register_tables)){
-    for (register_table_name in names(list_register_tables[[filter]])) {
-      register_table <- list_register_tables[[filter]][[register_table_name]]
-      render_register_json(filter, register_table, register_table_name)
+    # For the case of venues we have a nested list
+    if (filter == "venues"){
```

Comment:
I am not sure if this is the best check for "nested lists". Can you think of another way?
Or would we have to list all filters that can have subfilters here (and in many other places) ?

Alternatively: Would it be possible to nest the "non-venue" cases so that the same loop works for them, too?

I'm just trying to wrap my head around this code, and I see this pattern (check for venues, then two nested loops, otherwise one loop) several times in the code. If this is the best way for now - fine! I just would like you to double check. Thanks!

Reply:
An alternative I thought of is having the structure of the list as such:
list_register_tables <- list(
"venues" = list("category1_venue2" = table, "category_2_venue2" = table),
"codecheckers" = list("orcid_id1" = table, "orcid_id2" = table)
)

That way the structure for each filtered list is the same and I will create a helper function that reads in the venue register table name "category1_venue2" to obtain the venue category and the venue name. 

This will also remove the "filter subcat" argument in a bunch of functions and be replaced a conditional to check if filter=="venues" and handle that case differently.

Lmk what you think of this? If it's okay with you I will go ahead with it

Reply:
I'm more happy with that solution, but am also wondering if the approach to filtering based on the names of items in a list is really like things work well in R. You sort of mimic the behaviour of having a data structure with a two-piece key and a table as the value. You use this two-piece key to then iterate through all pages that need to be generated (the "unpacking" I comment on below).

Maybe it's worth taking a look if a tidy data structure and the corresponding filter mechanisms could help?

See https://tidyr.tidyverse.org/articles/tidy-data.html

You can also nest data.frames/tables: https://tidyr.tidyverse.org/articles/nest.html

On filtering and selecting: https://r4ds.hadley.nz/functions.html#sec-embracing

Please take a look these docs and see if that resonates with your though model of the data that you use here. If yes, then we can merge this PR as is and work on a possible rewrite filtering/passing the data in a separate issue. If no, then we go ahead with your preferred option (as is, or the alternative suggested above).

Reply:
Thanks for the sources. I have been looking into tidyR and getting myself acquainted with it and indeed it is better to use that and its `nest()` function to filter the data. 

I am working on redoing the code to use that method instead. Best making the changes now than later. Will ask you for a review again once I have finished that.